packages/core/test/integration/model/findAll.test.js

```diff
@@ -1396,6 +1396,34 @@ The following associations are defined on "Worker": "ToDos"`);
         }
       });
 
+      it('should allow a literal as VIRTUAL', async function () {
```

Comment:
This does not need to be an integration test. Could you change this test to be done using `queryGenerator.selectQuery` & only test the generated query?

You can see examples of that in `test/unit/sql/select-query.ts`, but this test could go in `test/unit/data-types/virtual.test.ts`

Reply:
I've been trying to do this but I think it is findAll the method that invokes _injectDependentVirtualAttributes and _expandAttributes and I need those to get the column definition instead of just the name

if I do: 
```js
const User = sequelize.define('User', {
        active: {
          type: DataTypes.VIRTUAL(DataTypes.BOOLEAN, includeAs => [
        dialectName === 'db2'
        ? literal(`(SELECT "createdAt" > CURRENT DATE - 7 DAYS FROM "Users" WHERE "User"."id" = "${includeAs.replace(/`/g, '')}"."id")`)
        : dialectName === 'sqlite' ? literal(`(SELECT "createdAt" > CURRENT_DATE - 7 * 24 * 60 * 60 * 1000 FROM Users WHERE "User"."id" = ${includeAs}."id")`)
        : dialectName === 'postgres' ? literal(`(SELECT "createdAt" > NOW() - INTERVAL '1 week' FROM "Users" WHERE "User"."id" = "${includeAs.replace(/`/g, '')}"."id")`)
        : dialectName === 'mssql' ? literal(`(SELECT COUNT(*) FROM Users WHERE Users.id = ${includeAs.replace(/`/g, '')}.id AND createdAt > DATEADD(WEEK, -1, GETDATE()))`)
        : literal(`(SELECT createdAt > CURRENT_DATE - 7 * 24 * 60 * 60 * 1000 FROM Users WHERE User.id = ${includeAs}.id)`),
        'active',
          ]),
        },
      });

const sql = queryGenerator.selectQuery(User.table, {
        model: User,
        attributes: ['active'],
      }, User);
```
      
 I get SELECT `active` FROM `Users` AS `User`;

Reply:
You're correct. A possible solution is to `spy` `sequelize.queryRaw` to see what the SQL looks like & prevent its execution, but still use the model method